docs/definition.rst

```diff
@@ -245,6 +260,38 @@ The following keys are valid for this section:
       The system packages to be installed, in bindep format. This may either
       be a filename, or a list of requirements (see below for an example).
 
+    ``exclude``
+      A list of Python or system requirements to be excluded from the top-level dependency requirements
+      of referenced collections. These exclusions will not apply to the user supplied Python or system
+      dependencies, nor will they apply to dependencies of dependencies (top-level only). Python dependency
+      exclusions should be a list of package names appearing under the ``python`` key name. System dependency
+      exclusions should be a list of system package names appearing under the ``system`` key name. If you
+      want to exclude *all* Python and system dependencies from one or more collections, supply the list
+      of collection names under the ``all_from_collections`` key.
+
+      The exclusion string should be the simple name of the requirement you want excluded. For example,
+      if you need to exclude the system requirement that appears as ``foo [!platform:gentoo]`` within
+      an included collection, then your exclusion string should be ``foo``. To exclude the Python
+      requirement ``bar == 1.0.0``, your exclusion string would be ``bar``.
```

Comment:
I question if using the "simple" name for exclusion is something we really want. I'm open to discussion on this.